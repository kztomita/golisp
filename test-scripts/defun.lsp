(defun plus (x y) (+ x y))

(print (plus 2 3))


(defun optional_arg (a &optional b (c 10) (d))
  (print (list a b c d))
  )

(optional_arg 1)
(optional_arg 1 2)
(optional_arg 1 2 3)
(optional_arg 1 2 3 4)
