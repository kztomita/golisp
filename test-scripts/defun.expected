5
"----"
(1 nil 10 nil)
(1 2 10 nil)
(1 2 3 nil)
(1 2 3 4)
"----"
(1 nil)
(1 (2))
(1 (2 3))
"----"
(1 nil nil)
(1 2 nil)
(1 2 (3))
(1 2 (3 4))
